osu.Game/Database/StandardisedScoreMigrationTools.cs

```diff
@@ -264,9 +266,63 @@ public static long ConvertFromLegacyTotalScore(ScoreInfo score, LegacyBeatmapCon
             switch (score.Ruleset.OnlineID)
             {
                 case 0:
+                    if (score.MaxCombo == 0 || score.Accuracy == 0)
+                    {
+                        return (long)Math.Round((
+                            0
+                            + 300000 * Math.Pow(score.Accuracy, 8)
+                            + bonusProportion) * modMultiplier);
+                    }
+
+                    // Assumption :
+                    // - sliders and slider-ticks are uniformly spread arround the beatmap
+                    //      thus we can ignore them without losing much precision (consider a map of hit-circles only !)
+                    // - the Ok/Meh hit results are uniformly spread in the score
+                    //      thus we can simplify and consider each hit result to be score.Accuracy without losing much precision
+                    // What is strippedV1/strippedV3 :
+                    // This is the ComboScore of v1/v3 were we remove all (map-)constant multipliers and accuracy multipliers (including hit results),
+                    // based on the previous assumptions. For Scorev1, this is basically the sum of squared combos (because without sliders: object_count == combo).
+                    double maxStrippedV1 = Math.Pow(maximumLegacyCombo, 2);
+                    double maxStrippedV3 = Math.Pow(maximumLegacyCombo, 1 + ScoreProcessor.COMBO_EXPONENT);
+
+                    double strippedV1 = maxStrippedV1 * comboProportion / score.Accuracy;
+
+                    double strippedV1FromMaxCombo = Math.Pow(score.MaxCombo, 2);
+                    double strippedV3FromMaxCombo = Math.Pow(score.MaxCombo, 1 + ScoreProcessor.COMBO_EXPONENT);
+
+                    // Compute approximate lower estimate scorev3 for that play
+                    // That is, a play were we made biggest amount of big combos (Repeat MaxCombo + 1 remaining big combo)
+                    // And didn't combo anything in the reminder of the map
+                    double possibleMaxComboRepeat = Math.Floor(strippedV1 / strippedV1FromMaxCombo);
```

Comment:
What is this even doing? What is this number?

```
strippedV1 = maxStrippedV1 * comboProportion / score.Accuracy = Math.Pow(maximumLegacyCombo, 2) * comboProportion / score.Accuracy
strippedV1FromMaxCombo = Math.Pow(score.MaxCombo, 2)
```

What possible significance could it have to divide these two numbers? It _feels_ like this is trying to calculate how many times the max combo is allowed to repeat in a play, but surely the combo portion, the accuracy, and the squares would make trying to calculate this from those two numbers completely incorrect? Why is this not just `Math.Floor(maximumLegacyCombo / score.MaxCombo)`?

Reply:
This is the maximum amount of repeats of the MaxCombo that are possible without exceeding the score of the player. Yes this would depend on the accuracy, and things like distribution of sliders in the map, but we are ignoring that and assuming the simple case. In reality, this should be a good approximation.